ext/drx/drx_time_scale.c

```diff
@@ -325,9 +364,13 @@ event_filter_syscall(void *drcontext, int sysnum)
 #ifndef X64
     case SYS_timer_gettime64:
     case SYS_timer_settime64:
+    case SYS_clock_nanosleep_time64:
 #endif
     case SYS_setitimer:
-    case SYS_getitimer: return true;
+    case SYS_getitimer:
+    case SYS_nanosleep:
```

Comment:
So we weren't intercepting and scaling nanosleep before? And this wasn't detected by unit tests?

Reply:
It looks like the unit test's use of syscall() causes the syscall instruction to be in a separate block from the number and so DR can't figure out the number and so can't ignore the syscall.

All the tests on large apps were with drmemtrace which asks to see every syscall.

Yes, this is disturbing. Maybe this fix should be separated out? I thought at first someone else was filtering when I noticed the filter missing but it seems that it would not scale for many app uses.

Reply:
Separated out into #7599 just to have that bug fix separate from this feature.